File: Code/Server/src/Core/Entities/ComentarioEquipo.cs
using System.ComponentModel.DataAnnotations.Schema;
using IMT_Reservas.Server.Core.Abstraction;

namespace IMT_Reservas.Server.Core.Entities;

public class ComentarioEquipo : Entity
{
    public int IdGrupoEquipo { get; set; }
    public int? IdComentarioPadre { get; set; }
    public string CarnetUsuario { get; set; } = string.Empty;
    public string Contenido { get; set; } = string.Empty;
    public DateTime FechaCreacion { get; set; } = DateTime.UtcNow;
    public int Likes { get; set; }
    public string LikedBy { get; set; } = string.Empty;

    [ForeignKey("IdGrupoEquipo")]
    public GrupoEquipo? GrupoEquipo { get; set; }

    [ForeignKey("IdComentarioPadre")]
    public ComentarioEquipo? ComentarioPadre { get; set; }

    [ForeignKey("CarnetUsuario")]
    public Usuario? Usuario { get; set; }
}

File: Code/Client/src/app/entities/equipment-group/model/comentario-equipo.ts
export type ComentarioEquipoOrden = 'recientes' | 'antiguos' | 'likes';

export interface ComentarioEquipo {
  id: number;
  idGrupoEquipo: number;
  idComentarioPadre: number | null;
  carnetUsuario: string;
  nombreUsuario: string;
  contenido: string;
  fechaCreacion: string;
  likes: number;
  likedByCurrentUser: boolean;
  puedeEliminar: boolean;
  respuestas: ComentarioEquipo[];
}
